pkg/descheduler/policyconfig_test.go

```diff
@@ -0,0 +1,443 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package descheduler
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	utilpointer "k8s.io/utils/pointer"
+	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/api/v1alpha1"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/defaultevictor"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/nodeutilization"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removeduplicates"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removefailedpods"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodshavingtoomanyrestarts"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatinginterpodantiaffinity"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatingnodeaffinity"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatingnodetaints"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint"
+	"sigs.k8s.io/descheduler/pkg/utils"
+	"sigs.k8s.io/descheduler/test"
+)
+
+func TestV1alpha1ToV1alpha2(t *testing.T) {
+	defaultEvictorPluginConfig := api.PluginConfig{
+		Name: defaultevictor.PluginName,
+		Args: &defaultevictor.DefaultEvictorArgs{
+			PriorityThreshold: &api.PriorityThreshold{
+				Value: utilpointer.Int32(utils.SystemCriticalPriority),
+			},
+		},
+	}
+	defaultEvictorPluginSet := api.PluginSet{
+		Enabled: []string{defaultevictor.PluginName},
+	}
+	type testCase struct {
+		description string
+		policy      *v1alpha1.DeschedulerPolicy
+		err         error
+		result      *api.DeschedulerPolicy
+	}
+	testCases := []testCase{
```

Comment:
You will not like me but would you please create another test which wires all existing strategy params into their corresponding arguments? So we can avoid https://github.com/kubernetes-sigs/descheduler/pull/961.

Reply:
Would be easier to write the tests and it would be less clunky onto the upper abstraction. We could test exactly what is being passed and what is coming out from params to args